spec/models/cart_spec.rb

```diff
@@ -1,26 +1,89 @@
 require 'rails_helper'
 
 
-  describe 'Class Methods' do
-    it '.add_item' do
-      cart = Cart.new({"1" => 2})
-      cart.add_item(1)
-      cart.add_item(9)
+describe 'instance methods' do
+  it '.add_item should add an item' do
+    merchant = create(:merchant)
+    item_1 = create(:item, user: merchant)
+    item_2 = create(:item, user: merchant)
+    cart = Cart.new(Hash.new(0))
+    cart.add_item(item_1)
+    cart.add_item(item_1)
+    cart.add_item(item_1)
+    cart.add_item(item_2)
 
-      expect(cart.contents).to eq({"1" => 3, "9" => 1})
-    end
+    expect(cart.contents).to eq({"#{item_1.id}" => 3, "#{item_2.id}" => 1})
+  end
 
-    it '.amount' do
-      cart = Cart.new({"1" => 2, "15" => 30})
+  it '.amount should return amount for each item' do
+    merchant = create(:merchant)
+    item_1 = create(:item, user: merchant)
+    item_2 = create(:item, user: merchant)
+    item_3 = create(:item, user: merchant)
 
-      expect(cart.amount(15)).to eq(30)
-      expect(cart.amount(1)).to eq(2)
-      expect(cart.amount(20)).to eq(0)
-    end
+    cart = Cart.new(Hash.new(0))
+    cart.add_item(item_1)
+    cart.add_item(item_1)
+    cart.add_item(item_1)
+    cart.add_item(item_2)
+    cart.add_item(item_3)
+    cart.add_item(item_3)
+
+    expect(cart.item_amount(item_1)).to eq(3)
+    expect(cart.item_amount(item_2)).to eq(1)
+    expect(cart.item_amount(item_3)).to eq(2)
+  end
+
+  it '.total_count should return total count of all items' do
+    cart = Cart.new({"1" => 2, "15" => 30})
+
+    expect(cart.total_count).to eq(32)
+  end
 
-    it '.total_count' do
-      cart = Cart.new({"1" => 2, "15" => 30})
+  it '.all_items should return all item objects in cart' do
+    merchant = create(:merchant)
+    item_1 = create(:item, user: merchant)
+    item_2 = create(:item, user: merchant)
+    cart = Cart.new(Hash.new(0))
+    cart.add_item(item_1)
+    cart.add_item(item_1)
+    cart.add_item(item_1)
+    cart.add_item(item_2)
+    all_items_in_cart = ([item_1, item_2])
 
-      expect(cart.total_count).to eq(32)
-    end 
+    expect(cart.all_items).to eq(all_items_in_cart)
+  end
+  describe 'it should return a subtotal for each item' do
+    it '.item_subtotal' do
+      merchant = create(:merchant)
+      item_1 = create(:item, user: merchant, price: 5)
+      item_2 = create(:item, user: merchant, price: 15)
+      cart = Cart.new(Hash.new(0))
+      cart.add_item(item_1)
```

Comment:
seeing a lot of multiple lines of cart.add_item, what if #add_item had a argument of count that defaults to 1 but that way if someone wanted to add multiple of the same item at once we wouldn't need some kind of loop or anything. This is definitely fine, but some food for thought.